stacks-signer/src/runloop.rs

```diff
@@ -90,6 +105,10 @@ pub struct RunLoop<C> {
     pub state: State,
     /// Wether mainnet or not
     pub mainnet: bool,
+    /// Observed blocks that we have seen so far
+    pub blocks: HashMap<Vec<u8>, BlockInfo>,
+    /// Transactions that we expect to see in the next block
```

Comment:
Does the order matter?  If not, then perhaps a `HashSet` might be better (especially since there can't be duplicates anyway)?

Reply:
Good point. Will update.